### PR TITLE
Knit TypeScript plugin

### PR DESCRIPTION
**Pull request by srikrsna-buf**
The TypeScript plugin for Knit.

### CODE REVIEW

File: plugins/bufbuild/knit-ts/v0.0.1/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# syntax=docker/dockerfile:1.4
+FROM node:18.15.0-alpine3.17 AS build
```
Comment (mfridman) [inline plugins/bufbuild/knit-ts/v0.0.1/Dockerfile:2]:
```suggestion
FROM node:18.16.0-alpine3.17 AS build
```

File: plugins/bufbuild/knit-ts/v0.0.1/buf.plugin.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+version: v1
+name: buf.build/bufbuild/knit-ts
+plugin_version: v0.0.1
+source_url: https://github.com/bufbuild/knit-ts
+description: Generates Knit schemas needed to use the Knit client in web browsers and Node.js.
+output_languages:
+  - typescript
+registry:
+  npm:
```
Comment (mfridman) [inline plugins/bufbuild/knit-ts/v0.0.1/buf.plugin.yaml:9]: Does it have any runtime `deps`?

Comment (srikrsna-buf) [reply plugins/bufbuild/knit-ts/v0.0.1/buf.plugin.yaml:9]: None

**Review by mfridman — APPROVED** [review]

File: plugins/bufbuild/knit-ts/v0.0.1/buf.plugin.yaml (hunk 2 above)
Comment (pkwarren) [reply plugins/bufbuild/knit-ts/v0.0.1/buf.plugin.yaml:9]: The demo depends on `@bufbuild/knit` - is that not needed?

https://github.com/bufbuild/knit-demo/blob/8766e187d5730da22783a22c65a398cbd90d7f21/ts/package.json#L13-L15

Comment (srikrsna-buf) [reply plugins/bufbuild/knit-ts/v0.0.1/buf.plugin.yaml:9]: The generated code doesn't import it, this should be only if the generated code uses it right?

Comment (pkwarren) [reply plugins/bufbuild/knit-ts/v0.0.1/buf.plugin.yaml:9]: Yep.

**Pull request merged**